girder/stemserver_plugin/models/stemimage.py

```diff
@@ -165,11 +165,22 @@ def all_frames(self, id, user):
         def _stream():
             nonlocal f
             with h5py.File(f, 'r') as rf:
+                current_size = 0
+                max_chunk_size = 64000
                 arrays = rf[path][()].tolist()
+                data = []
                 for i, array in enumerate(arrays):
-                    arrays[i] = array.tolist()
+                    array_size = array.size * array.dtype.itemsize
+                    if len(data) != 0:
+                        if current_size + array_size > max_chunk_size:
+                            yield msgpack.packb(data, use_bin_type=True)
+                            data = []
+                            current_size = 0
 
-                yield msgpack.packb(arrays, use_bin_type=True)
+                    data.append(array.tolist())
+                    current_size += array_size
+
+                yield msgpack.packb(data, use_bin_type=True)
```

Comment:
The data streaming above was probably the trickiest to do, since it is an array of variable length arrays. But I think this does something like we need, and it can be unpacked using something like this:
```
res = gc.get('/stem_images/%s/frames' % stem_image_id, jsonResp=False)

unpacker = msgpack.Unpacker(raw=False)
unpacker.feed(res.content)
frame_events = []
for val in unpacker:
    frame_events += val
```

Let me know if any changes are needed.